File: src/main/java/com/scottscmo/ppt/Presentation.java
package com.scottscmo.ppt;

import java.io.File;
import java.io.FileInputStream;
import java.io.FileOutputStream;
import java.io.IOException;
import java.util.List;

import org.apache.poi.xslf.usermodel.XMLSlideShow;
import org.apache.poi.xslf.usermodel.XSLFSlide;

public class Presentation {
    final static List<TemplateHandler> templateHandlers = List.of(
        new BibleTemplateHandler(),
        new SongTemplateHandler());

    public static void createFromTemplate(String templateFilePath, String outputFilePath) throws IOException {
        try (FileInputStream inStream = new FileInputStream(new File(templateFilePath))) {
            XMLSlideShow ppt = new XMLSlideShow(inStream);
            List<XSLFSlide> slides = ppt.getSlides();

            for (int i = 0; i < slides.size(); i++) {
                for (TemplateHandler templateHandler : templateHandlers) {
                    templateHandler.evaluateTemplate(ppt, slides.get(i).getSlideNumber() - 1);
                }
            }

            // write new PPTX
            try (FileOutputStream outStream = new FileOutputStream(new File(outputFilePath))) {
                ppt.write(outStream);
            }
            ppt.close();
        }
    }

    public static void main(String[] args) throws IOException {
        createFromTemplate("template.pptx", "example.pptx");
    }
}

File: src/main/kotlin/com/scottscmo/ppt/Presentation.kt
package com.scottscmo.ppt

import org.apache.poi.xslf.usermodel.XMLSlideShow
import java.io.File
import java.io.FileInputStream
import java.io.FileOutputStream
import java.io.IOException

object Presentation {
    private val bibleTemplateHandler = BibleTemplateHandler()
    private val songTemplateHandler = SongTemplateHandler()

    @Throws(IOException::class)
    fun createFromTemplate(templateFilePath: String, outputFilePath: String) {
        FileInputStream(File(templateFilePath)).use { inStream ->
            val ppt = XMLSlideShow(inStream)
            val slides = ppt.slides
            for (i in slides.indices) {
                bibleTemplateHandler.evaluateTemplate(ppt, slides[i].slideNumber - 1)
                songTemplateHandler.evaluateTemplate(ppt, slides[i].slideNumber - 1)
            }
            FileOutputStream(File(outputFilePath)).use { outStream -> ppt.write(outStream) }
            ppt.close()
        }
    }
}